tools/BaseImageUpdate/baseImage.ps1

```diff
@@ -74,4 +95,107 @@ function Update-ARM-BaseImages
         Set-Content $file.Path
     }
+}
+
+
+function Update-AMD64-BaseImages
+{
+    [CmdletBinding()]
+    param (
+        <# 
+        The new version of ASP .NET Core. 
+        This version is applied to both 'alpine' and 'nanoserver'
+        Ex: The new ASP .NET Core tag is 3.1.15-alpine3.13, the $NewASPNetCoreVersion = 3.1.15
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $NewASPNetCoreVersion,
+
+        <# 
+        The new version of Alpine base image. 
+        This version is only applied to 'alpine' images
+        Ex: The new ASP .NET Core tag is 3.1.15-alpine3.13, the $NewAlpineVersion = 3.13
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $NewAlpineVersion,
+
+        <# 
+        Object array contain file paths to base image dockerfile
+        #>
+        [Parameter(Mandatory=$false)]
+        [Object[]]
+        $FileLocale
+    )
+
+    <# REMARK: This function does not update 'debian' and 'azure function' base images!!! #>
+
+    if ($FileLocale.Count -gt 0)
+    {
+        $fileLocale = $FileLocale
+    }
+    else
+    {
+        $fileLocale = Get-Dockerfile-Locations
+    }
+
+    # Replace the underlying ASP .Net Core to the new version for 'nanoserver'
+    $baseAspNetLocale = $($($fileLocale | Convert-Path) -like "*\windows\amd64\*" -notlike "*\bin\*" | Resolve-path)
+    foreach ($file in $baseAspNetLocale)
+    {
+        # Note: The following dockerfile(s) are not automatically updated by this script
+        #    \iotedge\edge-modules\functions\samples\docker\windows\amd64\Dockerfile_______(Azure Function)
+        #    \iotedge\tools\snitch\snitcher\docker\windows\amd64\Dockerfile _______________(debian10)
+        (Get-Content -Encoding utf8 $file.Path) |
+        Foreach-Object { $_ -replace "ARG base_tag=.*.-nanoserver-1809", "ARG base_tag=$NewASPNetCoreVersion-nanoserver-1809" } |
+        Set-Content -Encoding utf8 $file.Path 
+    }
+
+    # Replace the underlying ASP .Net Core to the new version for 'alpine'
+    $baseAspNetLocale = $($($fileLocale | Convert-Path) -like "*\linux\amd64\*" -notlike "*\bin\*" | Resolve-path)
+    foreach ($file in $baseAspNetLocale)
+    {
+        # Note: The following dockerfile(s) are not automatically updated by this script
+        #    \iotedge\edge-modules\functions\samples\docker\linux\amd64\Dockerfile ________(Azure Function)
+        #    \iotedge\tools\snitch\prep-mail\docker\linux\amd64\Dockerfile ________________(Deprecating)
+        #    \iotedge\tools\snitch\snitcher\docker\linux\amd64\Dockerfile _________________(Deprecating)
+        #    \iotedge\edgelet\iotedged\docker\linux\amd64\Dockerfile ______________________(debian10)
+        (Get-Content -Encoding utf8 $file.Path) |
+        Foreach-Object { $_ -replace "ARG base_tag=.*.-alpine[\d]+\.[\d]+", "ARG base_tag=$NewASPNetCoreVersion-alpine$NewAlpineVersion" } |
```

Comment:
You could probably do all 3 replacements in 1 loop, since a file that has "nanoserver" is unlikely to have "alpine" and the rest of the loop looks the same.